src/main/groovy/edu/wpi/first/gradlerio/wpi/dependencies/WPIJsonDepsPlugin.groovy

```diff
@@ -0,0 +1,339 @@
+
+    static class JsonDependency {
+        String name
+        String version
+        String uuid
+        String[] mavenUrls
+        String jsonUrl
+        String fileName
+        JavaArtifact[] javaDependencies
+        JniArtifact[] jniDependencies
+        CppArtifact[] cppDependencies
+    }
+
+    static class WPIJsonDepsExtension {
+        List<JsonDependency> dependencies = []
+        final List<DelegatedDependencySet> nativeDependenciesList = []
+
+        final Project project
+
+        WPIJsonDepsExtension(Project project) {
+            this.project = project
+        }
+    }
+
+    @CompileDynamic
+    private JsonDependency constructJsonDependency(Object slurped) {
+        try {
+            return new JsonDependency(slurped)
+        } catch (def e) {
+            return null
+        }
+    }
+
+    @Override
+    void apply(Project project) {
+        project.pluginManager.apply(NativeDepsPlugin)
+        def wpi = project.extensions.getByType(WPIExtension)
+
+        def jsonExtension = project.extensions.create('wpiJsonDeps', WPIJsonDepsExtension, project)
+
+        def nativeclassifier = (
+                OperatingSystem.current().isWindows() ?
+                        System.getProperty("os.arch") == 'amd64' ? 'windowsx86-64' : 'windowsx86' :
+                        OperatingSystem.current().isMacOsX() ? "osxx86-64" :
+                                OperatingSystem.current().isLinux() ? "linuxx86-64" :
+                                        null
+        )
+
+        def jsonDepFolder = project.file('vendordeps')
+        JsonSlurper slurper = new JsonSlurper()
+
+        // Try to load dependencies JSON files
+        if (jsonDepFolder.exists()) {
+            jsonDepFolder.eachFileMatch FileType.FILES,~/.*\.json/, { File file ->
+                file.withReader {
+                    def slurped = slurper.parse(it)
+                    def dep = constructJsonDependency(slurped)
+                    if (dep == null) {
+                        //TODO Display an error
+                    } else {
+                        jsonExtension.dependencies << dep
+                    }
+                }
+            }
+        }
+
+         DependencySpecExtension dse = project.extensions.getByType(DependencySpecExtension)
+
+        // Add all URLs from dependencies
+        jsonExtension.dependencies.each { JsonDependency dep ->
+            dep.mavenUrls.each { url ->
+                project.repositories.maven { MavenArtifactRepository repo ->
+                    repo.url = url
+                }
+            }
+        }
+
+        project.configurations.maybeCreate('wpiJsonJni')
+
+        project.extensions.add('javaVendorLibraries', { String... ignoreLibraries ->
+            def returns = []
+            if (jsonExtension.dependencies != null) {
+                jsonExtension.dependencies.find { (!ignoreLibraries.contains(it.name) && !ignoreLibraries.contains(it.uuid)) } each { JsonDependency dep ->

```

Comment:
This can be reduced into
```groovy
returns += jsonExtension.dependencies.findAll { (!ignoreLibraries.contains(it.name) && !ignoreLibraries.contains(it.uuid)) }.collectMany { JsonDependency dep ->
   dep.javaDependencies.collect { JavaArtifact art -> "${art.groupId}:${art.artifactId}:${art.version}" }
}
```

Likewise for others